modules/collect/src/main/java/com/opengamma/strata/collect/io/ArrayByteSource.java

```diff
@@ -149,21 +155,32 @@ public static ArrayByteSource ofUtf8(String str) {
 
   //-------------------------------------------------------------------------
   /**
-   * Creates an instance from another byte source.
+   * Obtains an instance from another byte source.
    * 
    * @param other  the other byte source
    * @return the byte source
    * @throws UncheckedIOException if an IO error occurs
    */
   public static ArrayByteSource from(ByteSource other) {
     if (other instanceof ArrayByteSource) {
-      return (ArrayByteSource) other;
+      return ((ArrayByteSource) other);
+    }
+    String fileName = other instanceof BeanByteSource ? ((BeanByteSource) other).getFileName().orElse(null) : null;
```

Comment:
Probably want to nest this into the other `if/else` otherwise we're calling the `.toString()` when it isn't required